core/src/main/java/org/ldaptive/ext/MergeOperation.java

```diff
@@ -279,21 +283,60 @@ protected Result modify(
         Collections.addAll(resultModifications, modifications);
       }
       if (!resultModifications.isEmpty()) {
+        // if batching attribute values, create new attribute modifications as necessary
+        if (request.getAttributeValuesBatchSize() > 0) {
+          final List<AttributeModification> attrValuesModifications = new ArrayList<>(resultModifications.size());
+          for (AttributeModification am : resultModifications) {
+            if (request.getAttributeValuesBatchSize() < am.getAttribute().size()) {
+              divideList(
+                new ArrayList<>(am.getAttribute().getBinaryValues()),
+                request.getAttributeValuesBatchSize(),
+                values -> attrValuesModifications.add(
+                  new AttributeModification(
+                    am.getOperation(),
+                    LdapAttribute.builder().name(am.getAttribute().getName()).binaryValues(values).build())));
+            } else {
+              attrValuesModifications.add(am);
+            }
+          }
+          if (!attrValuesModifications.equals(resultModifications)) {
+            resultModifications.clear();
+            resultModifications.addAll(attrValuesModifications);
+          }
+        }
         logger.info(
           "Modifying target entry {} with modifications {} from source entry {} for request {}",
           target,
           resultModifications,
           source,
           request);
 
-        final ModifyOperation operation = modifyOperation != null ?
-          ModifyOperation.copy(modifyOperation) : new ModifyOperation();
-        operation.setConnectionFactory(connectionFactory);
-        final Result result = operation.execute(
-          ModifyRequest.builder()
-            .dn(target.getDn())
-            .modifications(resultModifications.toArray(AttributeModification[]::new))
-            .build());
+        Result result = null;
+        // if batching modifications, execute a modify operation for each batch
+        if (request.getModificationBatchSize() > 0 && request.getModificationBatchSize() < resultModifications.size()) {
+          final List<List<AttributeModification>> batchedResultModifications =
+            new ArrayList<>(resultModifications.size());
+          divideList(resultModifications, request.getModificationBatchSize(), batchedResultModifications::add);
+          for (List<AttributeModification> batch : batchedResultModifications) {
+            final ModifyOperation operation = modifyOperation != null ?
```

Comment:
I'm not crazy about cloning the operation to provide batching behavior, but I see that operations invoking other operations is an established pattern so it's probably nothing of concern especially since it's fairly clearly documented in the class javadocs. In general I would think an operation is self-contained, but it appears modify is a special case.

Reply:
The clone of the modify is carried over from the last patch, it's not related to this work. The `MergeOperation` carries properties for each add, delete, modify to allow custom configuration to be applied to each operation.

Reply:
All good.